pkg/runtime/subscription/subscription.go

```diff
@@ -175,6 +186,24 @@ func New(opts Options) (*Subscription, error) {
 				diag.DefaultComponentMonitoring.PubsubIngressEvent(ctx, name, strings.ToLower(string(contribpubsub.Retry)), "", msgTopic, 0)
 				return err
 			}
+
+			// fallback to message metadata to propagate the tracing information
+			if _, ok := cloudEvent[contribpubsub.TraceIDField]; !ok {
+				if traceid, ok := msg.Metadata[contribpubsub.TraceIDField]; ok {
+					cloudEvent[contribpubsub.TraceIDField] = traceid
+				}
+			}
+			if _, ok := cloudEvent[contribpubsub.TraceParentField]; !ok {
+				if traceparent, ok := msg.Metadata[contribpubsub.TraceParentField]; ok {
+					cloudEvent[contribpubsub.TraceIDField] = traceparent
```

Comment:
This seems to be different than the update logic above, where the TraceID populates both TraceID and TraceParent. In this block TraceParent is used for TraceID. Is this intentional?

Reply:
The `traceparent` override is based on [this comment](https://github.com/dapr/dapr/blob/master/pkg/runtime/pubsub/cloudevents.go#L51). I've made a change to make them consistent and adjusted the unit test